sqlair.go

```diff
@@ -464,24 +468,34 @@ func (db *DB) Begin(ctx context.Context, opts *TXOptions) (*TX, error) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
-	sqltx, err := db.sqldb.BeginTx(ctx, opts.plainTXOptions())
-	return &TX{sqltx: sqltx, db: db}, err
+	sqlconn, err := db.sqldb.Conn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	sqltx, err := sqlconn.BeginTx(ctx, opts.plainTXOptions())
+	return &TX{sqltx: sqltx, sqlconn: sqlconn, db: db}, err
 }
 
 // Commit commits the transaction.
 func (tx *TX) Commit() error {
 	if err := tx.setDone(); err != nil {
 		return err
 	}
-	return tx.sqltx.Commit()
+	if err := tx.sqltx.Commit(); err != nil {
+		return err
```

Comment:
The connection needs to be closed, whether there are errors or not.

Reply:
Closing of connection deferred at the start.